### PR TITLE
fix image rendering, use paste provider

### LINKED ISSUE #229625: Delay when pasting in large image.
Testing #229263

I can't consistently reproduce this, but there was a multi-second delay that i'd noticed twice when pasting in a large file (not sure if size was the issue, but that's the only time i'd noticed it).

The photo was:
Image (7680x4320)

In this case I dragged and dropped into the main chat panel

### PR DESCRIPTION
**Pull request by justschen**
this moves `chatImagePaste` content into `chatPasteProvider` which is invoked on paste.

fixes https://github.com/microsoft/vscode/issues/229625 and https://github.com/microsoft/vscode-copilot/issues/8637

cc. @benibenj 

paste:

https://github.com/user-attachments/assets/1627b7fd-53ef-4534-b13e-d8624d3180b5

dnd:

https://github.com/user-attachments/assets/79547ead-36bb-4dcf-be74-b9b444e8c372

### CODE REVIEW

File: src/vs/workbench/contrib/chat/browser/chatPasteProviders.ts [hunk 1]
```diff
@@ -12,26 +12,102 @@ import { ITextModel } from '../../../../editor/common/model.js';
...
+		const currentContextIds = new Set(Array.from(inputPart.attachedContext).map(context => context.id));
+		const filteredContext = [];
+
+		if (!currentContextIds.has(imageContext.id)) {
+			currentContextIds.add(imageContext.id);
+			filteredContext.push(imageContext);
+		}
+
+		inputPart.attachContext(false, ...filteredContext);
 		return;
```
Comment (benibenj) [inline src/vs/workbench/contrib/chat/browser/chatPasteProviders.ts:63]:
```suggestion
		const widget = this.chatWidgetService.getWidgetByInputUri(_model.uri);
		if (!widget) {
			return;
		}

		const attachContribution = widget.getContrib<ChatContextAttachments>(ChatContextAttachments.ID);
		attachContribution?.setContext(false, imageContext);
```

**Review by benibenj — COMMENTED** [review]
FYI, in this case you can use the `ChatContextAttachments` contribution which also takes care of duplicates. We are not able to use it for chat dnd sadly.

File: src/vs/workbench/contrib/chat/browser/chatPasteProviders.ts [hunk 2]
```diff
@@ -12,26 +12,96 @@ import { ITextModel } from '../../../../editor/common/model.js';
...
+	) { }
 
 	async provideDocumentPasteEdits(_model: ITextModel, _ranges: readonly IRange[], dataTransfer: IReadonlyVSDataTransfer, context: DocumentPasteContext, token: CancellationToken): Promise<DocumentPasteEditsSession | undefined> {
-		// TODO: @justschen setup image data types
-		// const entry = dataTransfer.get('image/png') || dataTransfer.get('image/jpeg') || dataTransfer.get('image/gif') || dataTransfer.get('image/webp');
+		if (!this.configurationService.getValue<boolean>('chat.experimental.imageAttachments')) {
+			return;
+		}
+
+		const currClipboard = await this.clipboardService.readImage();
```
Comment (mjbvz) [inline src/vs/workbench/contrib/chat/browser/chatPasteProviders.ts:40]: Can you use the `dataTransfer` here instead? It should have the image data in it

File: src/vs/workbench/contrib/chat/browser/chatPasteProviders.ts [hunk 3]
```diff
@@ -12,26 +12,96 @@ import { ITextModel } from '../../../../editor/common/model.js';
...
+		if (!imageContext) {
+			return;
+		}
+		const widget = this.chatWidgetService.getWidgetByInputUri(_model.uri);
+		if (!widget) {
+			return;
+		}
+
+		const attachContribution = widget.getContrib<ChatContextAttachments>(ChatContextAttachments.ID);
+		attachContribution?.setContext(false, imageContext);
```
Comment (mjbvz) [inline src/vs/workbench/contrib/chat/browser/chatPasteProviders.ts:55]: `provideDocumentPasteEdits` should not have any side effects. Maybe ok as a temporary workaround for now but let's look into allowing workspace edits to add attachments to chat. That way undo/redo will work nicely too

**Review by mjbvz — CHANGES_REQUESTED** [review]

**Review by mjbvz — DISMISSED** [review]
Approving for continued iteration but please make sure to follow up about using workspace edits

**Review by rzhao271 — APPROVED** [review]

**Pull request merged**